assets/js/theme/custom/mega-menu-search.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Open mega menu on search button click
+ */
+export default function megaMenuSearch() {
+    const MOBILE_WIDTH = 768
+    const megaMenu = document.querySelector('.js-navigation-megamenu')
+    const searchButton = document.querySelector('.js-mega-menu-search')
+
+    const openDesktopMenu = () => {
+        megaMenu.previousElementSibling.classList.toggle('is-open')
+        megaMenu.classList.toggle('is-open')
+    }
+
+    const updateEventListeners = () => {
+        const windowWidth = window.innerWidth
+
+        if (searchButton) {
+            if (windowWidth < MOBILE_WIDTH) {
+                searchButton.removeEventListener('click', openDesktopMenu)
+            } else {
+                searchButton.addEventListener('click', openDesktopMenu)
```

Comment:
If you are calling `updateEventListeners` on `resize` wouldn't it results with adding multiple (the same) click events attached to searchButton?

Reply:
up